src/Driver/DiscordRoleFilter.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/**
+ * This file is part of SeAT Discord Connector.
+ *
+ * Copyright (C) 2021  Troyburn <1537309279@character.id.eve.ccpgames.com>
+ *
+ * SeAT Discord Connector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * SeAT Discord Connector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+
+namespace Warlof\Seat\Connector\Drivers\Discord\Driver;
+
+class DiscordRoleFilter
+{
+    /**
+     *
+     * @var array of string
+     */
+    private $filter_specs;
+
+    /**
+     *
+     * @var bool
+     */
+    private $everyrole;
+
+    /**
+     *
+     * @var string EVERYROLE_ENTRY
+     */
+    public const EVERYROLE_ENTRY = '@@everyrole';
+
+    /**
+     * Arbitrary string "RoleName Example2 RemoveMe" exist to self document a valid setting in database.
+     *
+     * @var string DEFAULT_CAN_REMOVE_ROLES
+     */
+    public const DEFAULT_CAN_REMOVE_ROLES = '@@everyrole:RoleName Example2 RemoveMe';
```

Comment:
IIRC the design goal is not to break peoples systems during an upgrade.

An empty value does not make the overall module (seat-discord-connector) work as most users might expect, the empty value will disable functionality.

Not sure tooltip information alone helps.  I'm sure the tooltip information can be improved, I didn't spend much time on it (since I know what it does lol).

This is about a fresh install or an upgrade to this version and then opening the discord settings and ensuring the new fields are populated with defaults that continue to maintain old behavior where possible.  In the case of this new feature that should be possible, hence value is in code as well as documentation (README).

If you only put information in documentation and tooltips many people won't read/notice during upgrade and the first time they save the settings after upgrade (which could be months) their connector will stop working like it did before pressing Discord Settings -> Save.

How many support issues might this cause, due to not reading documentation around the uprade ?

The `@@everyrole` is an illegal role name.
That role names in discord can contain spaces.
That the use of `*` it also a valid role name in discord.
Discord has a `@everyone` with special meaning, so it wasn't a far stretch to use `@@everyrole`
I'm trying to convery to someone not reading the seat-discord-connector documentation and making a guess how to configure seat-discord-connector with 2 or more roles and when roles contain spaces all at the same time.  i.e. space is not the role name delimiter, space is allowed.  So ideally your bust guess will be correct first time.